types/prosemirror-keymap/index.d.ts

```diff
@@ -41,12 +43,23 @@ import { EditorView } from 'prosemirror-view';
  * which they appear determines their precedence (the ones early in
  * the array get to dispatch first).
  */
-export function keymap(bindings: { [key: string]: any }): Plugin;
+export function keymap<S extends Schema = any>(bindings: {
+  [key: string]: (
+    state: EditorState<S>,
+    dispatch: (tr: Transaction<S>) => void,
+    view: EditorView<S>
+  ) => boolean;
+}): Plugin;
+
 /**
  * Given a set of bindings (using the same format as
  * [`keymap`](#keymap.keymap), return a [keydown
  * handler](#view.EditorProps.handleKeyDown) handles them.
  */
-export function keydownHandler(bindings: {
-  [key: string]: any;
-}): (view: EditorView, event: Event) => boolean;
+export function keydownHandler<S extends Schema = any>(bindings: {
+  [key: string]: (
```

Comment:
Optional: Reduce repetitiveness by declaring and exporting a new interface

```typescript
interface Command {
  (
    state: EditorState<S>,
    dispatch: (tr: Transaction<S>) => void,
    view: EditorView<S>
  ) => boolean;
}
```

in `prosemirror-commands` and using it here.